### PR TITLE
fix for #2747

### LINKED ISSUE #2747: [Bug] Remove the null check for .WithAzureRegion()
When a null value is passed in the .WithAzureRegion() api, we point skip region discovery and use the global endpoint and null is a valid value for this api.

Remove the null check made here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/ee3cd9c2fe59c0c34ff3a65142bc16f017fd7b27/src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs#L206

### PR DESCRIPTION
**Pull request by jmprieur**
Fixes #2747

Taking this one as I want to quickly update the 1P doc, and I want the product to work the way it's declared to work.
(was a partner request)

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs [hunk 1]
```diff
@@ -203,11 +204,6 @@ public ConfidentialClientApplicationBuilder WithClientAssertion(Func<string> cli
         /// <returns>The builder to chain the .With methods</returns>
         public ConfidentialClientApplicationBuilder WithAzureRegion(string azureRegion = ConfidentialClientApplication.AttemptRegionDiscovery)
         {
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs:206]: there should be a test that failed? right?

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs (hunk 1 above)
Comment (jennyf19) [reply src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs:206]: did you check that we have a test which passes in null?

Comment (jmprieur) [reply src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs:206]: yes, I was surprized that no test failed, right.
But I'll add one ...

File: src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs [hunk 2]
```diff
@@ -203,11 +204,6 @@ public ConfidentialClientApplicationBuilder WithClientAssertion(Func<string> cli
         /// <returns>The builder to chain the .With methods</returns>
         public ConfidentialClientApplicationBuilder WithAzureRegion(string azureRegion = ConfidentialClientApplication.AttemptRegionDiscovery)
```
Comment (henrik-me) [inline src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs:205]: >azureRegion [](http://example.com/codeflow?start=75&length=11)

I would have assumed we should fail if null is used. This way people can believe they are using regional but they are not.

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs:205]: I'm pretty sure this fails with an `ArgumentNullException`

**Review by henrik-me — CHANGES_REQUESTED** [review]
:clock1:

File: src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs (hunk 2 above)
Comment (jmprieur) [reply src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs:205]: So, @henrik-me, shall I close this PR and the associated bug ?

Comment (jennyf19) [reply src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs:205]: but i don't see a test checking for this? unless i missed it.

**Comment by jmprieur** [comment]
Proposing to close this PR.

I've proposed the following update to the 1P doc:

`.WithRegion(region)`, doesn't accept `null`. If you want to conditionally enable region detection, you'll want to use the following pattern:
  
  ```CSharp
  var confidentialClientApplicationBuilder = ConfidentialClientApplicationBuilder(client_id)
                  .WithCertificate(x509cert);
  if (region != null)
  {
    confidentialClientApplicationBuilder.WithAzureRegion(region);
  }
  s_cca = confidentialClientApplicationBuilder.Build();
  ```

@henrik-me @neha-bhargava : do we agree?

**Pull request closed without merging**